install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
-  if [ "${arch}" = "x64" ] && [ "$(getconf LONG_BIT)" -eq 32 ]; then
+  if [ "$arch" = "x64" ] && [ "$(getconf LONG_BIT)" = "32" ]; then
     arch=i386
   fi
 
-  echo "${arch}"
-}
+  case $arch in
+    x86) arch="i386" ;;
+    i686) arch="i386" ;;
+    i386) arch="i386" ;;
+    aarch64) arch="arm64" ;;
+    armv5*) arch="armv5" ;;
+    armv6*) arch="armv6" ;;
+    armv7*) arch="armv7" ;;
+  esac
 
-confirm() {
-  if [ -z "${FORCE-}" ]; then
-    printf "%s " "${MAGENTA}?${NO_COLOR} $* ${BOLD}[y/N]${NO_COLOR}"
-    set +e
-    read -r yn </dev/tty
-    rc=$?
-    set -e
-    if [ $rc -ne 0 ]; then
-      error 'Error reading from prompt (please re-run with the `--yes` option)'
-      exit 1
-    fi
-    if [ "$yn" != "y" ] && [ "$yn" != "yes" ]; then
-      error 'Aborting (please answer "yes" to continue)'
-      exit 1
-    fi
-  fi
+  echo $arch
 }
 
-check_bin_dir() {
-  local bin_dir="$1"
+arch_check() {
+  arch=${arch:-"$(uname_arch)"}
+  case $arch in
+    x86_64) return 0 ;;
+  esac
+
+  platform=$(uname_platform)
+
+  error "$arch builds are not yet available for Starship"
+  printf "\n" >&2
+  info "If you would like to see a build for your configuration,"
+  info "please create an issue requesting a build for ${MAGENTA}${arch}-${platform}${NO_COLOR}:"
+  info "${BOLD}${UNDERLINE}https://github.com/starship/starship/issues/new/${NO_COLOR}"
+  return 1
+}
 
+bin_dir_check() {
   if [ ! -d "$BIN_DIR" ]; then
     error "Installation location $BIN_DIR does not appear to be a directory"
+    printf "\n" >&2
     info "Make sure the location exists and is a directory, then try again."
-    exit 1
+    return 1
   fi
 
-  # https://stackoverflow.com/a/11655875
-  local good
-  good=$(
-    IFS=:
-    for path in $PATH; do
-      if [ "${path}" = "${bin_dir}" ]; then
-        echo 1
-        break
-      fi
-    done
-  )
+  return 0
+}
 
-  if [ "${good}" != "1" ]; then
-    warn "Bin directory ${bin_dir} is not in your \$PATH"
-  fi
+path_check() {
+  # https://stackoverflow.com/a/9663359
+  case :$PATH: in
+    *:$BIN_DIR:*) return 0 ;;
+  esac
+
+  warn "Bin directory $BIN_DIR is not in your \$PATH"
+  printf "\n"
+
+  return 0
 }
 
-# defaults
-if [ -z "${PLATFORM-}" ]; then
-  PLATFORM="$(detect_platform)"
-fi
+confirm() {
+  printf "%s " "${MAGENTA}?${NO_COLOR} $* ${BOLD}[y/N]${NO_COLOR}"
+  {
+    read -r yn </dev/tty
+    rc=$?
+  } || true
 
-if [ -z "${BIN_DIR-}" ]; then
-  BIN_DIR=/usr/local/bin
-fi
+  if [ $rc != 0 ]; then
+    error "Error reading from prompt (please re-run with the \`--yes\` option)"
+    return 1
+  elif [ "$yn" != "y" ] && [ "$yn" != "yes" ]; then
+    error 'Aborting (please answer "yes" to continue)'
+    return 1
+  fi
 
-if [ -z "${ARCH-}" ]; then
-  ARCH="$(detect_arch)"
-fi
+  return 0
+}
 
-if [ -z "${BASE_URL-}" ]; then
-  BASE_URL="https://github.com/starship/starship/releases"
-fi
+# defaults
+PREFIX=${PREFIX:-"/usr/local"}
```

Comment:
Why separate out `PREFIX` and `BIN_DIR`? This has the potential to be a bit confusing when the thing you're asked by the prompt (install location) and the environment variable you set (`PREFIX`) are different.

Reply:
This is actually a convention in *nix world. https://www.gnu.org/prep/standards/html_node/Directory-Variables.html

Reply:
Okay, so at the moment, if I wanted to install starship to `DIR`, I would set `BIN_DIR=$DIR` because we don't install stuff to share/include/lib/libexec/whatever. I was thinking of `PREFIX` as the final install location.

Reply:
PREFIX is really just a prefix/parent dir. 🙃